### DISCUSSION TITLE
Auto save option for Notepadsapp

### DISCUSSION THREAD

**Discussion by Dincer146** [0x7c13/Notepads#1020] [Ideas]
I actually like your notepad app its better than the default one but i just want one more option and it's auto-save. When i turn off the computer and forget to save all of the stuff the changes that was made to the notepads are gone so it would be to great and if you can't make that feature you can detach the save in the close button so that we can close the notepad without having to press "save and close" button or it can automatically save the changes that made to the notepad when it closes.

**Comment by 0x7c13** [comment]
<img width="544" alt="Screen Shot 2022-01-10 at 2 04 01 PM" src="https://user-images.githubusercontent.com/1166162/148846314-b220a6f7-92d9-4ba6-83d0-9231f09a6e1a.png">

**Comment by Betaum1303** [comment]
Abrimos tantas abas não salvas no notepads que ele deveria guardar mais backups. Meu problema ocorreu ontem e eu já não tenho mais nada das minhas anotações feitas ao longo de mais de 6 meses.

**Reply by 0x7c13** [reply]
You should not rely on the session snapshot feature for "saving" your files. This is not how this feature is intended and implemented. Always backup your files. Not only you should backup your files but you should also consider cloud backup solutions or personal NAS etc.

**Comment by Betaum1303** [comment]
Isso é algo que aprendi da pior forma. Infelizmente perdi tudo que anotei nos últimos 6 meses. E isso é irrecuperável. Qualquer ocisa que eu faça a partir de hoje não vai me trazer todo esse conteúdo de volta. Perdi anotações de trabalho, teses e outras informações que eu iria usar. Vou voltar a notar tudo no Microsoft word. Ele tem salvamento automático.

**Reply by 0x7c13** [reply]
You can recover your deleted files using 3rd party software. Do not download new things or save new things to your system drive and you have a high chance to recover them.

**Reply by 0x7c13** [reply]
Search "disk file recovery software for windows"

**Reply by 0x7c13** [reply]
Or try Disk drill: https://www.disk-drill.com/ (I personally use this one so I would only recomment it)

**Reply by Betaum1303** [reply]
Instalei e estou rodando ele. Algum nome de arquivo específico para procurar? Não encontrei a pasta "19282JackieLiu.Notepads-Beta_echhpq9pdbte8" na lista de achados do DiskDrill

**Reply by 0x7c13** [reply]
Let it run for your C: drive and look for all the files without extensions. The session files are named with a GUID string + "-LastSaved" or "-Pending"

**Reply by Betaum1303** [reply]
Das mais de 30 abas que eu tinha, achei o arquivo de duas, as mais irrelevantes e bem desatualizado. Alguma outra string a pesquisar que não seja "-pending" e "-lastsaved"? Os arquivos podem ter algum outro nome ou indicativo?

**Reply by Betaum1303** [reply]
Algum outro repositório que não seja "19282JackieLiu.Notepads-Beta_echhpq9pdbte8" para procurar nos arquivos do diskdrill?

**Reply by 0x7c13** [reply]
Look for any files under C:\Users<your username>\AppData\Local\Packages\

**Reply by Betaum1303** [reply]
Tentei de tudo. Imagine que eu achei um arquivo de agosto deste ano mas não achei os arquivos que estava usando a uns 2 ou 3 dias atrás. Infelizmente, perdi as anotações de um ano de trabalho. Coisas que jamais vou conseguir reaver.

**Reply by 0x7c13** [reply]
Sorry to hear that, but I still don't understand why you keep writing things but not save them for more than 6 months? Are you "thinking" that the session snapshot feature is equal to "auto-save"? No offense here, just want to understand how you use the software. I have a PR committed yesterday to not delete files when Notepads is failed to recover from a failure state caused by filesystem or Windows. Hope it will help in the future. But again, sorry for your loss. PLEASE do remember to save your files and backup your files regularly. I personally do that as well because I know how painful it is when losing something important (I also lost my stuff before). Not only you should backup your important assets on your computer but also you should backup those to a cloud storage or another device since anything can happen, you know.

**Reply by Betaum1303** [reply]
Eu só acreditei que o aplicativo era confiável. Trabalho com uma advogada. Eu tinha uma aba para cada processo que cuidávamos. Ía anotando naquela aba as coisas relevantes sobre aquele processo. No final, quando ele era concluído, eu simplesmente apagava a aba. Aprendi da pior forma que não podia confiar no que estava ali. Eu tenho 3 backups de tudo o que tenho. Em 3 HDs diferentes, no entanto, como eram anotações, não vi necessidade de fazer backup delas. Daqui pra frente não confio em mais nada.

**Comment by pablo-s-dev** [comment]
Please add auto save bros

**Reply by Betaum1303** [reply]
I did. It worked for a few months and then notepads crash. When It returns all the anotations were gone. And there was no backup files. All lost.

**Reply by 0x7c13** [reply]
"Session snapshot" IS NOT "auto save". He is asking for "auto-save" feature.
